### DISCUSSION TITLE
How to use `getStaticProps` with dynamic API routes on Vercel

### DISCUSSION THREAD

**Discussion by bravokiloecho** [vercel/next.js#18191] [Help] [answered]
I have a next.js app running on Vercel. The problem I have is...

1. To use `getStaticProps` you have to use an absolute URL
2. In my `getStaticProps` I'm making a call to a dynamic API route to fetch some JSON
3. When the build process runs for the first time on Vercel the URL of the API route doesn't point to anywhere (because it hasn't been built yet). And,
4. On subsequent deploys, if I change the code in the API route, I can't access this during the build process that runs when I push this code.

How can I resolve these issues? Surely I'm missing something obvious.

**Comment by ljosberinn** [comment, marked as answer]
The docs mention [this](https://nextjs.org/docs/basic-features/data-fetching#simple-example-1) right **above** the link:

> Note: You should not use fetch() to call an API route in your application. Instead, directly import the API route and call its function yourself. You may need to slightly refactor your code for this approach.
> Fetching from an external API is fine!

**Reply by bravokiloecho** [reply]
Oh yes! Thank you so much for spotting that. Very good to know
